llama-rs/src/lib.rs

```diff
@@ -110,10 +110,15 @@ impl Default for InferenceParameters {
 type TokenId = i32;
 type Token = String;
 
-#[derive(Default)]
 pub struct Vocabulary {
-    /// Maps every integer (index) token id to its corresponding string
-    mapping: Vec<Token>,
+    /// Maps every integer (index) token id to its corresponding token
+    id_to_token: Vec<Token>,
```

Comment:
If this the ids and same order as the existing parser, it seems like it would be very easy to support both and just allow selecting between parsers with an option. That way there really wouldn't be a cost to merging this.

Reply:
We should actually bring the current tokenizer changes from llama.cpp, because both this and the old version of the tokenizer are probably wrong, so I think there's no need to keep both (and offer a configuration flag) when the user probably just wants the one that is correct.